### PR TITLE
Create intersection point and compute lightning

### PR DESCRIPTION
**Pull request by novalain**

### CODE REVIEW

**Comment by novalain** [comment]
Important!! Merge this **after** Material PR!

File: include/ray.h [hunk 1]
```diff
@@ -2,19 +2,21 @@
...
-
   ColorDbl color_;
-
-  void* intersecting_triangle_;
+  // TODO: make smart pointer
+  IntersectionPoint* intersection_point_ = nullptr;
 
 public:
   Ray() {}
+  ~Ray() { if(intersection_point_) delete intersection_point_; }
```
Comment (Carlbaum) [inline include/ray.h:19]: Personal preference: ALWAYS encapsule the scope of if's, for's and while's etc. with {}

File: src/camera.cc [hunk 2]
```diff
@@ -2,6 +2,7 @@
 #include "ray.h"
 #include "scene.h"
 #include "scene_object.h"
+#include "point_light.h"
```
Comment (Carlbaum) [inline src/camera.cc:5]: We need to refactor later, as it makes no sense to need point_light.h in the camera class

File: include/ray.h (hunk 1 above)
Comment (novalain) [reply include/ray.h:19]: Yes agree, I was just lazy

File: src/camera.cc (hunk 2 above)
Comment (novalain) [reply src/camera.cc:5]: class RaytraceManager?

Comment (novalain) [reply src/camera.cc:5]: btw point light shouldn't be needed here as scene contains vector of point lights. we should use those, this was just for testing

Comment (Carlbaum) [reply src/camera.cc:5]: Would probably be the best solution, yeah!

File: src/camera.cc [hunk 3]
```diff
@@ -96,20 +97,55 @@ void Camera::ClearColorBuffer(ColorDbl clear_color) {
...
+  // TODO: loop through lights in scene and remove hardcode
+  PointLight pl = PointLight(Vertex(0, 2, -2), 0.01f, COLOR_WHITE);
+  Direction light_direction = pl.get_position() - p.get_position();
+
+  float intensity = pl.get_intensity();
+  ColorDbl light_color = pl.get_color();
+  Direction L = glm::normalize(pl.get_position() - p.get_position());
+  Direction N = glm::normalize(p.get_normal());
+
+  ColorDbl result = glm::dot(L, N) * light_color * intensity * p.get_material().get_color();
```
Comment (Carlbaum) [inline src/camera.cc:118]: I wonder if this is truly correct. Multiplying 2 colors with each other would make the result f-ing huge. Say both are white i.e. 255^2.

File: include/material.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  float transparence_;
+  ColorDbl color_;
+
+public:
+  float get_diffuse() { return diffuse_; }
+  float get_specular() { return specular_; }
+  float get_transparence() { return transparence_; }
+  ColorDbl get_color() { return color_; }
+
+public:
```
Comment (Carlbaum) [inline include/material.h:19]: Public scope declared twice

**Pull request closed without merging**